scheduler/replica_scheduler.go

```diff
@@ -143,90 +165,82 @@ func (rcs *ReplicaScheduler) getDiskCandidates(nodeInfo map[string]*longhorn.Nod
 				// For empty zone label, we treat them as
 				// one zone.
 				usedZones[node.Status.Zone] = true
+				replicasCountPerNode[r.Spec.NodeID] = replicasCountPerNode[r.Spec.NodeID] + 1
+			}
+		}
+	}
+
+	filterNodesWithLessThanTwoReplicas := func(nodes map[string]*longhorn.Node) map[string]*longhorn.Node {
+		result := map[string]*longhorn.Node{}
+		for nodeName, node := range nodes {
+			if replicasCountPerNode[nodeName] < 2 {
+				result[nodeName] = node
 			}
 		}
+		return result
 	}
```

Comment:
Not quite understand this part's purpose.
Let's have a review meeting if possible.

Reply:
This is the major closure to help #3076 when node soft anti-affinity is disabled to make sure being able to find out available extra disks to schedule on the same node where replicas all are being evicted. 

```go
		if diskCandidates := getDiskCandidatesFromNodes(filterNodesWithLessThanTwoReplicas(nodesWithEvictingReplicas)); len(diskCandidates) > 0 {
			return diskCandidates
		}
```

Reply:
I see. Then I am still thinking these two cases https://github.com/longhorn/longhorn-manager/blob/f9ee08e2e391dbba9a8d5fe096d6d19c99c39588/scheduler/replica_scheduler.go#L222-L241

Reply:
Discussed this question with @jenting over Slack.

Reply:
Created an issue for unit test implementation https://github.com/longhorn/longhorn/issues/3253